tools/test_utils.py

```diff
@@ -374,3 +380,32 @@ def run_cpp_example_test(build_dir):
 
     # Return to the original directory
     os.chdir(root_pwd)
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument(
+        '--run_func',
+        type=str,
+        help="runs a function in this file\n",
+        action="store")
+    parser.add_argument(
+        'func_args',
+        metavar='args',
+        type=str,
+        nargs='*',
+        help='optional argument(s) for the function')
+
+    arguments = parser.parse_args()
+
+    if (arguments.run_func):
+        if arguments.run_func in globals():
+            method_to_call = globals()[arguments.run_func]
+            dict_func_args = dict.fromkeys(arguments.func_args, 1)
+            method_to_call(*(arguments.func_args))
+        else:
+            sys.exit("Cannot find function!")
+
+
+if __name__ == '__main__':
+    main()
```

Comment:
This is probably not needed if you implement the manifest parsing in the GTest runner.